q2doc/usage/validation.py

```diff
@@ -0,0 +1,12 @@
+import ast
+
+
+class BlockValidator(ast.NodeVisitor):
```

Comment:
A great resource for implementing an `ast.NodeVisitor`:  [Green Tree Snakes - the missing Python AST docs — Green Tree Snakes 1.0 documentation](https://greentreesnakes.readthedocs.io/en/latest/)